src/dawn/Optimizer/PassSetBoundaryCondition.cpp

```diff
@@ -0,0 +1,332 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/PassSetBoundaryCondition.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Optimizer/Stencil.h"
+#include "dawn/Optimizer/StencilInstantiation.h"
+#include "dawn/SIR/ASTExpr.h"
+#include "dawn/SIR/ASTStmt.h"
+#include "dawn/SIR/ASTUtil.h"
+#include "dawn/SIR/ASTVisitor.h"
+#include "dawn/Support/Assert.h"
+#include "dawn/Support/Logging.h"
+#include <iostream>
+#include <set>
+#include <unordered_map>
+#include <vector>
+
+namespace dawn {
+
+namespace {
+/// @brief This method analyzes the Extents in which a field with a given ID is used within a a
+/// Stencil. This is used to compute how big the halo exchanges need to be.
+/// @param s A shared ptr to the stencil to be analyzed
+/// @param ID the FieldID of the Field to be analized
+/// @return the full extent of the field in the stencil
+static dawn::Extents analyzeStencilExtents(const std::shared_ptr<Stencil>& s, int fieldID) {
+  Extents fullExtents;
+  Stencil& stencil = *s;
+
+  int numStages = stencil.getNumStages();
```

Comment:
there is no need for this declaration,
just use a for range
```
for(auto stage: stencil.getStages())
```

Reply:
stencil has no get-stages. We can either get the multistages and the stages from there or leave as is. Thoughts?

Reply:
we leave as is since iterators are coming and will replace this as soon as they arrive